projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DeploymentServiceV2.java

```diff
@@ -44,6 +51,56 @@ public class DeploymentServiceV2 {
   private final ActualJobDeploymentRepository actualJobDeploymentRepository;
   private final DataJobsKubernetesService dataJobsKubernetesService;
 
+  /**
+   * This method updates an existing job deployment in the database. Only fields present in the job
+   * deployment are updated, other fields are not overridden.
+   *
+   * @param dataJob the data job to which the deployment is associated
+   * @param jobDeployment the deployment to patch with
+   */
+  public void patchDesiredDbDeployment(
+      DataJob dataJob, JobDeployment jobDeployment, String userDeployer) {
+    actualJobDeploymentRepository
+        .findById(dataJob.getName())
+        .ifPresentOrElse(
+            oldDeployment -> {
+              var mergedDeployment =
+                  DeploymentModelConverter.mergeDeployments(
+                      oldDeployment, jobDeployment, userDeployer);
+              saveNewDesiredDeployment(mergedDeployment, dataJob);
+            },
+            () -> {
+              throw new DataJobDeploymentNotFoundException(dataJob.getName());
+            });
+  }
+
+  /**
+   * Create or update a deployment in the database. If the deployment already exists, behaves like
+   * patch
+   *
+   * @param dataJob The data job to which the deployment is associated.
+   * @param jobDeployment the new deployment.
+   * @param userDeployer the user.
+   */
+  public void updateDesiredDbDeployment(
+      DataJob dataJob, JobDeployment jobDeployment, String userDeployer) {
+    actualJobDeploymentRepository
+        .findById(dataJob.getName())
+        .ifPresentOrElse(
+            oldDeployment -> patchDesiredDbDeployment(dataJob, jobDeployment, userDeployer),
+            () -> {
+              var newDeployment =
+                  DeploymentModelConverter.toJobDeployment(userDeployer, jobDeployment);
+              saveNewDesiredDeployment(newDeployment, dataJob);
+            });
+  }
+
+  private void saveNewDesiredDeployment(DesiredDataJobDeployment deployment, DataJob dataJob) {
```

Comment:
As per the recent changes, you should set deployment.setUserInitiated(true) as well.